### ISSUE TITLE
Have some kind of release cycle for flows

### ISSUE THREAD

**Issue by WilliamTales** [rapidpro/rapidpro#827]
For many times I've faced users who managed to add a new category, rename an existing one or link it to another flow step after having a flow started to a contact group!
The thing is that once any modification is done on flow's rulesets ("Wait for response" like) all previous replies are lost. This is not a problem... in fact it's really important to match many Privacy Policy requirements.
But it will cause an unadvised flow editor to mess up a poll flow, to say the least!
Would be great to have a feature to turn on meaning that that flow is already in production and having real replies from contacts.

What I suggest is adding a checkbox like feature to leave it on developer position while creating the flow's steps and testing. Similar to Facebook developers apps.
And after a flow is set as like, the flow editor would pop up a message advising against editing that flow with an option to go on even so.

On dev:
![image](https://user-images.githubusercontent.com/35577716/49101729-ab9af000-f255-11e8-8751-f7da5a025cc5.png)

Flow is ready:
![image](https://user-images.githubusercontent.com/35577716/49101631-74c4da00-f255-11e8-8125-694e39d117b1.png)

Warning message:
![image](https://user-images.githubusercontent.com/35577716/49102873-b2773200-f258-11e8-90e0-aaa28728b215.png)

Appreciated.

**Comment by nicpottier** [comment]
I think this probably belongs as part of a broader conversation around modification of flows that already have contacts that have gone through it. IE, even the current behavior of changes immediately being "live" is problematic from the perspective that people may be in the flow or that you may not be able to test the flow fully before the change takes effect etc..

So I'd rather solve this in that context of what does a "release cycle" look like for a flow that is active long term. IE, maybe you copy the flow, allow editing of that, then at some point "commit" it as the active flow and at that point you could have warnings like this for example?

Definitely something we want to address and think about but that will have to wait for the new editor to launch.

As an aside, data isn't actually lost when you remove a node, the results are still present in the runs, they just aren't shown. You can "restore" (really just show again) the old results by reverting the flow to the state it was when those nodes existed.

For GDPR purposes the way you delete results is by deleting runs.

**Comment by nicpottier** [comment]
Renaming this issue to what I think is more relevant, that it may be nice to have a way of "editing" a flow before it becoming live. In an imaginary ideal world having some kind of unit test running on the flow before it can become live may be nice. In an even crazier imaginary word people may need to have those changes reviewed by an admin?

To be clear, not agreeing all those things belong, but recording them.
